### PR TITLE
Crier/Slack: Allow to overwrite the channel on the job

### PR DESCRIPTION
**Pull request by alvaroaleman**
With this PR it is possible to overwrite the Slack channel crier reports to on the job. This is useful when different teams own different jobs as it allows to only notify the team that owns the job.

Fixes https://github.com/kubernetes/test-infra/issues/12935

/assign @krzyzacy 

/cc @xrstf

### CODE REVIEW

**Comment by alvaroaleman** [comment]
@krzyzacy tests already succeeded after kicking

**Comment by krzyzacy** [comment]
yeah, was a bit suspicious with an empty log, probably GCS issue again.

Maybe worth to add an announcement for new features?

/lgtm
/hold

**Comment by alvaroaleman** [comment]
@krzyzacy An announcement is a great idea, adedd that

File: prow/pjutil/pjutil_test.go [hunk 1]
```diff
@@ -700,3 +702,44 @@ func TestCreateRefs(t *testing.T) {
...
+				return nil
+			},
+		},
+	}
+
+	for _, tc := range testCases {
+		t.Run(tc.name, func(t *testing.T) {
+			pj := specFromJobBase(tc.jobBase)
+			if err := tc.verify(pj); err != nil {
+				t.Fatalf("Verifycation failed: %v", err)
```
Comment (xrstf) [inline prow/pjutil/pjutil_test.go]: It's *Verification*, not *Verifycation*.

**Comment by alvaroaleman** [comment]
Okay, typos in tests are fixed now as well. Mind to re-lgtm @krzyzacy and can we remove the `/hold`, too?

**Pull request merged**